## Imports

In [43]:
import pandas as pd
from pathlib import Path

## Define Paths

In [44]:
ROOT_DIR = Path.cwd().parent
RAW_DIR = ROOT_DIR / "data" / "raw"

print(f"Project root: {ROOT_DIR}")
print(f"Raw data folder: {RAW_DIR}")

Project root: c:\Repos\capstone-hospital-quality
Raw data folder: c:\Repos\capstone-hospital-quality\data\raw


## Reusable Loader

In [45]:
def load_cms_file(file_name: str) -> pd.DataFrame:
    """Load a CMS file and detect whether it uses tabs or commas."""
    file_path = RAW_DIR / file_name

    if not file_path.exists():
        raise FileNotFoundError(f"File not found: {file_path}")

    with file_path.open("r", encoding="utf-8-sig") as file:
        header = file.readline()

    separator = "\t" if header.count("\t") > header.count(",") else ","

    df = pd.read_csv(
        file_path,
        sep=separator,
        dtype={"Facility ID": "string"},
        low_memory=False,
    )

    delimiter_name = "TAB" if separator == "\t" else "COMMA"
    print(f"{file_name}: {df.shape}, delimiter={delimiter_name}")

    return df

## Load all Datasets

In [46]:
hospital_general = load_cms_file("hospital_general_info.csv")
hcahps = load_cms_file("HCAHPS.csv")
hospital_infections = load_cms_file("hospital_infections.csv")
hospital_mortality = load_cms_file("hospital_mortality.csv")
hospital_readmissions = load_cms_file("hospital_readmissions.csv")

hospital_general_info.csv: (5432, 38), delimiter=TAB
HCAHPS.csv: (325856, 22), delimiter=TAB
hospital_infections.csv: (172512, 15), delimiter=TAB
hospital_mortality.csv: (95840, 18), delimiter=TAB
hospital_readmissions.csv: (67088, 20), delimiter=TAB


## Summary Helper

In [47]:
def dataset_summary(df: pd.DataFrame, name: str) -> None:
    """Display a basic exploration summary for a dataframe."""
    print("=" * 70)
    print(name)
    print("=" * 70)
    print(f"Rows: {df.shape[0]:,}")
    print(f"Columns: {df.shape[1]}")
    print()

    df.info()

    print("\nFirst Five Rows")
    display(df.head())

    print("\nDescriptive Statistics")
    display(df.describe(include="all").transpose())

    print("\nTop Missing-Value Counts")
    display(
        df.isna()
        .sum()
        .sort_values(ascending=False)
        .head(10)
        .to_frame("Missing Count")
    )

## Summarize all Datasets

In [48]:
datasets = {
    "Hospital General Information": hospital_general,
    "HCAHPS": hcahps,
    "Hospital Infections": hospital_infections,
    "Hospital Mortality": hospital_mortality,
    "Hospital Readmissions": hospital_readmissions,
}

for name, df in datasets.items():
    dataset_summary(df, name)

Hospital General Information
Rows: 5,432
Columns: 38

<class 'pandas.DataFrame'>
RangeIndex: 5432 entries, 0 to 5431
Data columns (total 38 columns):
 #   Column                                            Non-Null Count  Dtype  
---  ------                                            --------------  -----  
 0   Facility ID                                       5432 non-null   string 
 1   Facility Name                                     5432 non-null   str    
 2   Address                                           5432 non-null   str    
 3   City/Town                                         5432 non-null   str    
 4   State                                             5432 non-null   str    
 5   ZIP Code                                          5432 non-null   int64  
 6   County/Parish                                     5432 non-null   str    
 7   Telephone Number                                  5432 non-null   str    
 8   Hospital Type                                     5432 

,Facility ID,Facility Name,Address,City/Town,State,ZIP Code,County/Parish,Telephone Number,Hospital Type,Hospital Ownership,...,Count of READM Measures Better,Count of READM Measures No Different,Count of READM Measures Worse,READM Group Footnote,Pt Exp Group Measure Count,Count of Facility Pt Exp Measures,Pt Exp Group Footnote,TE Group Measure Count,Count of Facility TE Measures,TE Group Footnote
0,10001,SOUTHEAST HEALTH MEDICAL CENTER,1108 ROSS CLARK CIRCLE,DOTHAN,AL,36301,HOUSTON,(334) 793-8701,Acute Care Hospitals,Government - Hospital District or Authority,...,1,9,1,NaN,15,15,NaN,10,10,NaN
1,10005,MARSHALL MEDICAL CENTERS,2505 U S HIGHWAY 431 NORTH,BOAZ,AL,35957,MARSHALL,(256) 593-8310,Acute Care Hospitals,Government - Hospital District or Authority,...,1,8,0,NaN,15,15,NaN,10,10,NaN
2,10006,NORTH ALABAMA MEDICAL CENTER,1701 VETERANS DRIVE,FLORENCE,AL,35630,LAUDERDALE,(256) 768-8400,Acute Care Hospitals,Proprietary,...,1,8,0,NaN,15,15,NaN,10,9,NaN
3,10007,MIZELL MEMORIAL HOSPITAL,702 N MAIN ST,OPP,AL,36467,COVINGTON,(334) 493-3541,Acute Care Hospitals,Voluntary non-profit - Private,...,0,3,2,NaN,15,5,NaN,10,7,NaN
4,10011,ST. VINCENT'S EAST,50 MEDICAL PARK EAST DRIVE,BIRMINGHAM,AL,35235,JEFFERSON,(205) 838-3122,Acute Care Hospitals,Voluntary non-profit - Private,...,1,5,2,29.0,15,10,29.0,10,7,29.0



Descriptive Statistics


,count,unique,top,freq,mean,std,min,25%,50%,75%,max
Facility ID,5432,5432,10001,1,NaN,NaN,NaN,NaN,NaN,NaN,NaN
Facility Name,5432,5301,MEMORIAL HOSPITAL,12,NaN,NaN,NaN,NaN,NaN,NaN,NaN
Address,5432,5398,100 HOSPITAL DRIVE,7,NaN,NaN,NaN,NaN,NaN,NaN,NaN
City/Town,5432,3048,CHICAGO,34,NaN,NaN,NaN,NaN,NaN,NaN,NaN
State,5432,56,TX,468,NaN,NaN,NaN,NaN,NaN,NaN,NaN
ZIP Code,5432.0,NaN,NaN,NaN,53822.804124,27052.075419,603.0,32796.0,55087.0,76105.5,99929.0
County/Parish,5432,1555,LOS ANGELES,88,NaN,NaN,NaN,NaN,NaN,NaN,NaN
Telephone Number,5432,5393,(469) 341-7800,4,NaN,NaN,NaN,NaN,NaN,NaN,NaN
Hospital Type,5432,8,Acute Care Hospitals,3115,NaN,NaN,NaN,NaN,NaN,NaN,NaN
Hospital Ownership,5432,12,Voluntary non-profit - Private,2316,NaN,NaN,NaN,NaN,NaN,NaN,NaN



Top Missing-Value Counts


,Missing Count
TE Group Footnote,4422
READM Group Footnote,4208
MORT Group Footnote,4032
Pt Exp Group Footnote,3414
Safety Group Footnote,3280
Meets criteria for birthing friendly designation,3169
Hospital overall rating footnote,3111
State,0
ZIP Code,0
Address,0


HCAHPS
Rows: 325,856
Columns: 22

<class 'pandas.DataFrame'>
RangeIndex: 325856 entries, 0 to 325855
Data columns (total 22 columns):
 #   Column                                 Non-Null Count   Dtype 
---  ------                                 --------------   ----- 
 0   Facility ID                            325856 non-null  string
 1   Facility Name                          325856 non-null  str   
 2   Address                                325856 non-null  str   
 3   City/Town                              325856 non-null  str   
 4   State                                  325856 non-null  str   
 5   ZIP Code                               325856 non-null  int64 
 6   County/Parish                          325856 non-null  str   
 7   Telephone Number                       325856 non-null  str   
 8   HCAHPS Measure ID                      325856 non-null  str   
 9   HCAHPS Question                        325856 non-null  str   
 10  HCAHPS Answer Description              325856

,Facility ID,Facility Name,Address,City/Town,State,ZIP Code,County/Parish,Telephone Number,HCAHPS Measure ID,HCAHPS Question,...,Patient Survey Star Rating Footnote,HCAHPS Answer Percent,HCAHPS Answer Percent Footnote,HCAHPS Linear Mean Value,Number of Completed Surveys,Number of Completed Surveys Footnote,Survey Response Rate Percent,Survey Response Rate Percent Footnote,Start Date,End Date
0,10001,SOUTHEAST HEALTH MEDICAL CENTER,1108 ROSS CLARK CIRCLE,DOTHAN,AL,36301,HOUSTON,(334) 793-8701,H_COMP_1_A_P,"Patients who reported that their nurses ""Alway...",...,NaN,76,NaN,Not Applicable,952,NaN,17,NaN,7/1/2024,6/30/2025
1,10001,SOUTHEAST HEALTH MEDICAL CENTER,1108 ROSS CLARK CIRCLE,DOTHAN,AL,36301,HOUSTON,(334) 793-8701,H_COMP_1_SN_P,"Patients who reported that their nurses ""Somet...",...,NaN,6,NaN,Not Applicable,952,NaN,17,NaN,7/1/2024,6/30/2025
2,10001,SOUTHEAST HEALTH MEDICAL CENTER,1108 ROSS CLARK CIRCLE,DOTHAN,AL,36301,HOUSTON,(334) 793-8701,H_COMP_1_U_P,"Patients who reported that their nurses ""Usual...",...,NaN,18,NaN,Not Applicable,952,NaN,17,NaN,7/1/2024,6/30/2025
3,10001,SOUTHEAST HEALTH MEDICAL CENTER,1108 ROSS CLARK CIRCLE,DOTHAN,AL,36301,HOUSTON,(334) 793-8701,H_COMP_1_LINEAR_SCORE,Nurse communication - linear mean score,...,NaN,Not Applicable,NaN,90,952,NaN,17,NaN,7/1/2024,6/30/2025
4,10001,SOUTHEAST HEALTH MEDICAL CENTER,1108 ROSS CLARK CIRCLE,DOTHAN,AL,36301,HOUSTON,(334) 793-8701,H_COMP_1_STAR_RATING,Nurse communication - star rating,...,NaN,Not Applicable,NaN,Not Applicable,952,NaN,17,NaN,7/1/2024,6/30/2025



Descriptive Statistics


,count,unique,top,freq,mean,std,min,25%,50%,75%,max
Facility ID,325856,4792,10001,68,NaN,NaN,NaN,NaN,NaN,NaN,NaN
Facility Name,325856,4672,MEMORIAL HOSPITAL,816,NaN,NaN,NaN,NaN,NaN,NaN,NaN
Address,325856,4760,100 HOSPITAL DRIVE,408,NaN,NaN,NaN,NaN,NaN,NaN,NaN
City/Town,325856,2925,CHICAGO,2040,NaN,NaN,NaN,NaN,NaN,NaN,NaN
State,325856,56,TX,27608,NaN,NaN,NaN,NaN,NaN,NaN,NaN
ZIP Code,325856.0,NaN,NaN,NaN,54206.306553,26930.786614,603.0,33047.5,55779.5,76208.0,99929.0
County/Parish,325856,1547,LOS ANGELES,5168,NaN,NaN,NaN,NaN,NaN,NaN,NaN
Telephone Number,325856,4763,(469) 341-7800,272,NaN,NaN,NaN,NaN,NaN,NaN,NaN
HCAHPS Measure ID,325856,68,H_COMP_1_A_P,4792,NaN,NaN,NaN,NaN,NaN,NaN,NaN
HCAHPS Question,325856,68,"Patients who reported that their nurses ""Alway...",4792,NaN,NaN,NaN,NaN,NaN,NaN,NaN



Top Missing-Value Counts


,Missing Count
Patient Survey Star Rating Footnote,311312
HCAHPS Answer Percent Footnote,238697
Survey Response Rate Percent Footnote,209644
Number of Completed Surveys Footnote,209644
Facility ID,0
Facility Name,0
ZIP Code,0
State,0
City/Town,0
Address,0


Hospital Infections
Rows: 172,512
Columns: 15

<class 'pandas.DataFrame'>
RangeIndex: 172512 entries, 0 to 172511
Data columns (total 15 columns):
 #   Column                Non-Null Count   Dtype 
---  ------                --------------   ----- 
 0   Facility ID           172512 non-null  string
 1   Facility Name         172512 non-null  str   
 2   Address               172512 non-null  str   
 3   City/Town             172512 non-null  str   
 4   State                 172512 non-null  str   
 5   ZIP Code              172512 non-null  int64 
 6   County/Parish         172512 non-null  str   
 7   Telephone Number      172512 non-null  str   
 8   Measure ID            172512 non-null  str   
 9   Measure Name          172512 non-null  str   
 10  Compared to National  172512 non-null  str   
 11  Score                 172512 non-null  str   
 12  Footnote              79965 non-null   str   
 13  Start Date            172512 non-null  str   
 14  End Date              172512 non

,Facility ID,Facility Name,Address,City/Town,State,ZIP Code,County/Parish,Telephone Number,Measure ID,Measure Name,Compared to National,Score,Footnote,Start Date,End Date
0,10001,SOUTHEAST HEALTH MEDICAL CENTER,1108 ROSS CLARK CIRCLE,DOTHAN,AL,36301,HOUSTON,(334) 793-8701,HAI_1_CILOWER,Central Line Associated Bloodstream Infection ...,Better than the National Benchmark,0.073,NaN,7/1/2024,6/30/2025
1,10001,SOUTHEAST HEALTH MEDICAL CENTER,1108 ROSS CLARK CIRCLE,DOTHAN,AL,36301,HOUSTON,(334) 793-8701,HAI_1_CIUPPER,Central Line Associated Bloodstream Infection ...,Better than the National Benchmark,0.785,NaN,7/1/2024,6/30/2025
2,10001,SOUTHEAST HEALTH MEDICAL CENTER,1108 ROSS CLARK CIRCLE,DOTHAN,AL,36301,HOUSTON,(334) 793-8701,HAI_1_DOPC,Central Line Associated Bloodstream Infection:...,Better than the National Benchmark,9871,NaN,7/1/2024,6/30/2025
3,10001,SOUTHEAST HEALTH MEDICAL CENTER,1108 ROSS CLARK CIRCLE,DOTHAN,AL,36301,HOUSTON,(334) 793-8701,HAI_1_ELIGCASES,Central Line Associated Bloodstream Infection ...,Better than the National Benchmark,10.399,NaN,7/1/2024,6/30/2025
4,10001,SOUTHEAST HEALTH MEDICAL CENTER,1108 ROSS CLARK CIRCLE,DOTHAN,AL,36301,HOUSTON,(334) 793-8701,HAI_1_NUMERATOR,Central Line Associated Bloodstream Infection ...,Better than the National Benchmark,3,NaN,7/1/2024,6/30/2025



Descriptive Statistics


,count,unique,top,freq,mean,std,min,25%,50%,75%,max
Facility ID,172512,4792,10001,36,NaN,NaN,NaN,NaN,NaN,NaN,NaN
Facility Name,172512,4672,MEMORIAL HOSPITAL,432,NaN,NaN,NaN,NaN,NaN,NaN,NaN
Address,172512,4760,100 HOSPITAL DRIVE,216,NaN,NaN,NaN,NaN,NaN,NaN,NaN
City/Town,172512,2925,CHICAGO,1080,NaN,NaN,NaN,NaN,NaN,NaN,NaN
State,172512,56,TX,14616,NaN,NaN,NaN,NaN,NaN,NaN,NaN
ZIP Code,172512.0,NaN,NaN,NaN,54206.306553,26930.823346,603.0,33047.5,55779.5,76208.0,99929.0
County/Parish,172512,1547,LOS ANGELES,2736,NaN,NaN,NaN,NaN,NaN,NaN,NaN
Telephone Number,172512,4763,(469) 341-7800,144,NaN,NaN,NaN,NaN,NaN,NaN,NaN
Measure ID,172512,36,HAI_1_CILOWER,4792,NaN,NaN,NaN,NaN,NaN,NaN,NaN
Measure Name,172512,36,Central Line Associated Bloodstream Infection ...,4792,NaN,NaN,NaN,NaN,NaN,NaN,NaN



Top Missing-Value Counts


,Missing Count
Footnote,92547
Facility Name,0
Facility ID,0
City/Town,0
State,0
ZIP Code,0
Address,0
County/Parish,0
Telephone Number,0
Measure Name,0


Hospital Mortality
Rows: 95,840
Columns: 18

<class 'pandas.DataFrame'>
RangeIndex: 95840 entries, 0 to 95839
Data columns (total 18 columns):
 #   Column                Non-Null Count  Dtype 
---  ------                --------------  ----- 
 0   Facility ID           95840 non-null  string
 1   Facility Name         95840 non-null  str   
 2   Address               95840 non-null  str   
 3   City/Town             95840 non-null  str   
 4   State                 95840 non-null  str   
 5   ZIP Code              95840 non-null  int64 
 6   County/Parish         95840 non-null  str   
 7   Telephone Number      95840 non-null  str   
 8   Measure ID            95840 non-null  str   
 9   Measure Name          95840 non-null  str   
 10  Compared to National  95840 non-null  str   
 11  Denominator           95840 non-null  str   
 12  Score                 95840 non-null  str   
 13  Lower Estimate        95840 non-null  str   
 14  Higher Estimate       95840 non-null  str   
 15  Fo

,Facility ID,Facility Name,Address,City/Town,State,ZIP Code,County/Parish,Telephone Number,Measure ID,Measure Name,Compared to National,Denominator,Score,Lower Estimate,Higher Estimate,Footnote,Start Date,End Date
0,10001,SOUTHEAST HEALTH MEDICAL CENTER,1108 ROSS CLARK CIRCLE,DOTHAN,AL,36301,HOUSTON,(334) 793-8701,COMP_HIP_KNEE,Rate of complications for hip/knee replacement...,No Different Than the National Rate,27,3.2,1.7,5.9,NaN,4/1/2021,3/31/2024
1,10001,SOUTHEAST HEALTH MEDICAL CENTER,1108 ROSS CLARK CIRCLE,DOTHAN,AL,36301,HOUSTON,(334) 793-8701,Hybrid_HWM,Hybrid Hospital-Wide All-Cause Risk Standardiz...,No Different Than the National Rate,1835,4.5,2.6,7.4,NaN,7/1/2023,6/30/2024
2,10001,SOUTHEAST HEALTH MEDICAL CENTER,1108 ROSS CLARK CIRCLE,DOTHAN,AL,36301,HOUSTON,(334) 793-8701,MORT_30_AMI,Death rate for heart attack patients,No Different Than the National Rate,270,11.4,9.1,14.3,NaN,7/1/2021,6/30/2024
3,10001,SOUTHEAST HEALTH MEDICAL CENTER,1108 ROSS CLARK CIRCLE,DOTHAN,AL,36301,HOUSTON,(334) 793-8701,MORT_30_CABG,Death rate for CABG surgery patients,No Different Than the National Rate,144,3,1.6,5.8,NaN,7/1/2021,6/30/2024
4,10001,SOUTHEAST HEALTH MEDICAL CENTER,1108 ROSS CLARK CIRCLE,DOTHAN,AL,36301,HOUSTON,(334) 793-8701,MORT_30_COPD,Death rate for COPD patients,No Different Than the National Rate,112,9.4,6.4,13.6,NaN,7/1/2021,6/30/2024



Descriptive Statistics


,count,unique,top,freq,mean,std,min,25%,50%,75%,max
Facility ID,95840,4792,10001,20,NaN,NaN,NaN,NaN,NaN,NaN,NaN
Facility Name,95840,4672,MEMORIAL HOSPITAL,240,NaN,NaN,NaN,NaN,NaN,NaN,NaN
Address,95840,4760,100 HOSPITAL DRIVE,120,NaN,NaN,NaN,NaN,NaN,NaN,NaN
City/Town,95840,2925,CHICAGO,600,NaN,NaN,NaN,NaN,NaN,NaN,NaN
State,95840,56,TX,8120,NaN,NaN,NaN,NaN,NaN,NaN,NaN
ZIP Code,95840.0,NaN,NaN,NaN,54206.306553,26930.88579,603.0,33047.5,55779.5,76208.0,99929.0
County/Parish,95840,1547,LOS ANGELES,1520,NaN,NaN,NaN,NaN,NaN,NaN,NaN
Telephone Number,95840,4763,(469) 341-7800,80,NaN,NaN,NaN,NaN,NaN,NaN,NaN
Measure ID,95840,20,COMP_HIP_KNEE,4792,NaN,NaN,NaN,NaN,NaN,NaN,NaN
Measure Name,95840,20,Rate of complications for hip/knee replacement...,4792,NaN,NaN,NaN,NaN,NaN,NaN,NaN



Top Missing-Value Counts


,Missing Count
Footnote,50795
Facility ID,0
Facility Name,0
Address,0
State,0
City/Town,0
County/Parish,0
Telephone Number,0
Measure ID,0
ZIP Code,0


Hospital Readmissions
Rows: 67,088
Columns: 20

<class 'pandas.DataFrame'>
RangeIndex: 67088 entries, 0 to 67087
Data columns (total 20 columns):
 #   Column                       Non-Null Count  Dtype 
---  ------                       --------------  ----- 
 0   Facility ID                  67088 non-null  string
 1   Facility Name                67088 non-null  str   
 2   Address                      67088 non-null  str   
 3   City/Town                    67088 non-null  str   
 4   State                        67088 non-null  str   
 5   ZIP Code                     67088 non-null  int64 
 6   County/Parish                67088 non-null  str   
 7   Telephone Number             67088 non-null  str   
 8   Measure ID                   67088 non-null  str   
 9   Measure Name                 67088 non-null  str   
 10  Compared to National         67088 non-null  str   
 11  Denominator                  67088 non-null  str   
 12  Score                        67088 non-null  str   

,Facility ID,Facility Name,Address,City/Town,State,ZIP Code,County/Parish,Telephone Number,Measure ID,Measure Name,Compared to National,Denominator,Score,Lower Estimate,Higher Estimate,Number of Patients,Number of Patients Returned,Footnote,Start Date,End Date
0,10001,SOUTHEAST HEALTH MEDICAL CENTER,1108 ROSS CLARK CIRCLE,DOTHAN,AL,36301,HOUSTON,(334) 793-8701,EDAC_30_AMI,Hospital return days for heart attack patients,Fewer Days Than Average per 100 Discharges,273,-15.6,-25.9,-4.2,264,68,NaN,7/1/2021,6/30/2024
1,10001,SOUTHEAST HEALTH MEDICAL CENTER,1108 ROSS CLARK CIRCLE,DOTHAN,AL,36301,HOUSTON,(334) 793-8701,EDAC_30_HF,Hospital return days for heart failure patients,Average Days per 100 Discharges,652,-1.1,-10,7.9,541,186,NaN,7/1/2021,6/30/2024
2,10001,SOUTHEAST HEALTH MEDICAL CENTER,1108 ROSS CLARK CIRCLE,DOTHAN,AL,36301,HOUSTON,(334) 793-8701,EDAC_30_PN,Hospital return days for pneumonia patients,More Days Than Average per 100 Discharges,507,17.4,8.2,27.5,471,114,NaN,7/1/2021,6/30/2024
3,10001,SOUTHEAST HEALTH MEDICAL CENTER,1108 ROSS CLARK CIRCLE,DOTHAN,AL,36301,HOUSTON,(334) 793-8701,Hybrid_HWR,Hybrid Hospital-Wide All-Cause Readmission Mea...,No Different Than the National Rate,2824,15.1,13.2,17.2,Not Applicable,Not Applicable,NaN,7/1/2023,6/30/2024
4,10001,SOUTHEAST HEALTH MEDICAL CENTER,1108 ROSS CLARK CIRCLE,DOTHAN,AL,36301,HOUSTON,(334) 793-8701,OP_32,Rate of unplanned hospital visits after colono...,No Different Than the National Rate,234,12.7,9.5,17,Not Applicable,Not Applicable,NaN,1/1/2022,12/31/2024



Descriptive Statistics


,count,unique,top,freq,mean,std,min,25%,50%,75%,max
Facility ID,67088,4792,10001,14,NaN,NaN,NaN,NaN,NaN,NaN,NaN
Facility Name,67088,4672,MEMORIAL HOSPITAL,168,NaN,NaN,NaN,NaN,NaN,NaN,NaN
Address,67088,4760,100 HOSPITAL DRIVE,84,NaN,NaN,NaN,NaN,NaN,NaN,NaN
City/Town,67088,2925,CHICAGO,420,NaN,NaN,NaN,NaN,NaN,NaN,NaN
State,67088,56,TX,5684,NaN,NaN,NaN,NaN,NaN,NaN,NaN
ZIP Code,67088.0,NaN,NaN,NaN,54206.306553,26930.946005,603.0,33047.5,55779.5,76208.0,99929.0
County/Parish,67088,1547,LOS ANGELES,1064,NaN,NaN,NaN,NaN,NaN,NaN,NaN
Telephone Number,67088,4763,(469) 341-7800,56,NaN,NaN,NaN,NaN,NaN,NaN,NaN
Measure ID,67088,14,EDAC_30_AMI,4792,NaN,NaN,NaN,NaN,NaN,NaN,NaN
Measure Name,67088,14,Hospital return days for heart attack patients,4792,NaN,NaN,NaN,NaN,NaN,NaN,NaN



Top Missing-Value Counts


,Missing Count
Footnote,34496
Facility ID,0
Address,0
Facility Name,0
City/Town,0
State,0
County/Parish,0
ZIP Code,0
Measure ID,0
Measure Name,0


## Verify the Merge Key

In [49]:
for name, df in datasets.items():
    print(f"\n{name}")
    print(f"Facility ID present: {'Facility ID' in df.columns}")

    if "Facility ID" in df.columns:
        print(f"Rows: {len(df):,}")
        print(f"Unique Facility IDs: {df['Facility ID'].nunique():,}")
        print(f"Missing Facility IDs: {df['Facility ID'].isna().sum():,}")
        print(f"Duplicate Facility IDs: {df['Facility ID'].duplicated().sum():,}")


Hospital General Information
Facility ID present: True
Rows: 5,432
Unique Facility IDs: 5,432
Missing Facility IDs: 0
Duplicate Facility IDs: 0

HCAHPS
Facility ID present: True
Rows: 325,856
Unique Facility IDs: 4,792
Missing Facility IDs: 0
Duplicate Facility IDs: 321,064

Hospital Infections
Facility ID present: True
Rows: 172,512
Unique Facility IDs: 4,792
Missing Facility IDs: 0
Duplicate Facility IDs: 167,720

Hospital Mortality
Facility ID present: True
Rows: 95,840
Unique Facility IDs: 4,792
Missing Facility IDs: 0
Duplicate Facility IDs: 91,048

Hospital Readmissions
Facility ID present: True
Rows: 67,088
Unique Facility IDs: 4,792
Missing Facility IDs: 0
Duplicate Facility IDs: 62,296


## Review Column Names

In [50]:
for name, df in datasets.items():
    print(f"\n{name}")
    for column in df.columns:
        print(column)


Hospital General Information
Facility ID
Facility Name
Address
City/Town
State
ZIP Code
County/Parish
Telephone Number
Hospital Type
Hospital Ownership
Emergency Services
Meets criteria for birthing friendly designation
Hospital overall rating
Hospital overall rating footnote
MORT Group Measure Count
Count of Facility MORT Measures
Count of MORT Measures Better
Count of MORT Measures No Different
Count of MORT Measures Worse
MORT Group Footnote
Safety Group Measure Count
Count of Facility Safety Measures
Count of Safety Measures Better
Count of Safety Measures No Different
Count of Safety Measures Worse
Safety Group Footnote
READM Group Measure Count
Count of Facility READM Measures
Count of READM Measures Better
Count of READM Measures No Different
Count of READM Measures Worse
READM Group Footnote
Pt Exp Group Measure Count
Count of Facility Pt Exp Measures
Pt Exp Group Footnote
TE Group Measure Count
Count of Facility TE Measures
TE Group Footnote

HCAHPS
Facility ID
Facility Name


## Check for missing values

In [51]:
for name, df in datasets.items():
    not_available_counts = (
        df.astype("string")
        .eq("Not Available")
        .sum()
        .sort_values(ascending=False)
    )

    not_available_counts = not_available_counts[not_available_counts > 0]

    print(f"\n{name}: 'Not Available' values")
    display(not_available_counts.to_frame("Count"))


Hospital General Information: 'Not Available' values


,Count
Hospital overall rating,2250
Count of Facility Safety Measures,2078
Count of Safety Measures Better,2078
Count of Safety Measures No Different,2078
Count of Safety Measures Worse,2078
Count of Facility Pt Exp Measures,1957
Count of MORT Measures Better,1327
Count of MORT Measures Worse,1327
Count of Facility MORT Measures,1327
Count of MORT Measures No Different,1327



HCAHPS: 'Not Available' values


,Count
Number of Completed Surveys,56848
Survey Response Rate Percent,56848
HCAHPS Answer Percent,51932
Patient Survey Star Rating,14544
HCAHPS Linear Mean Value,12928



Hospital Infections: 'Not Available' values


,Count
Compared to National,104808
Score,76325



Hospital Mortality: 'Not Available' values


,Count
Higher Estimate,43781
Score,43781
Lower Estimate,43781
Denominator,41906
Compared to National,31893



Hospital Readmissions: 'Not Available' values


,Count
Score,31857
Lower Estimate,31857
Higher Estimate,31857
Denominator,31857
Compared to National,15988
Number of Patients,6041
Number of Patients Returned,6041


## Create Inventory Table

In [52]:
inventory = []

for name, df in datasets.items():
    inventory.append(
        {
            "Dataset": name,
            "Rows": df.shape[0],
            "Columns": df.shape[1],
            "Facility ID Present": "Facility ID" in df.columns,
            "Unique Facility IDs": (
                df["Facility ID"].nunique()
                if "Facility ID" in df.columns
                else pd.NA
            ),
        }
    )

inventory_df = pd.DataFrame(inventory)
display(inventory_df)

,Dataset,Rows,Columns,Facility ID Present,Unique Facility IDs
0,Hospital General Information,5432,38,True,5432
1,HCAHPS,325856,22,True,4792
2,Hospital Infections,172512,15,True,4792
3,Hospital Mortality,95840,18,True,4792
4,Hospital Readmissions,67088,20,True,4792


## Findings
-Five CMS datasets were successfully loaded
-Hospital General Information contains 5432 hospitals and 38 variables
-All datasets contain the Facility ID field (target) needed for merging
-Several variables contain missing value or "Not Available" entries that required cleaning

## Next Steps
-Standardize variables and prepare data for merging

##